content/blog/options/index.md

```diff
@@ -0,0 +1,237 @@
+
+```rust
+// Assume that this fetches the user from somewhere
+fn get_user() -> Option<String> {
+    None
+}
+
+fn get_user_name() -> Result<String> {
+    let user = get_user()?;
+    // Do something with `user`
+    Ok(user)
+}
+```
+
+Almost everyone has received the following dreaded error message:
+
+```rust
+error[E0277]: the `?` operator can only be used on `Result`s, not `Option`s, in a function that returns `Result`
+  --> src/lib.rs:10:26
+   |
+9  | fn get_user_name() -> Result<String> {
+   | ------------------------------------ this function returns a `Result`
+10 |     let user = get_user()?;
+   |                          ^ use `.ok_or(...)?` to provide an error compatible with `std::result::Result<String, Box<dyn std::error::Error>>`
+   |
+   = help: the trait `FromResidual<Option<Infallible>>` is not implemented for `std::result::Result<String, Box<dyn std::error::Error>>`
+   = help: the following other types implement trait `FromResidual<R>`:
+             <std::result::Result<T, F> as FromResidual<Yeet<E>>>
+             <std::result::Result<T, F> as FromResidual<std::result::Result<Infallible, E>>>
+```
+
+([Rust Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7001ff6af6c0bcbf44249691d65b086f))
+
+Ouch. This is scary-looking.
+There's a lot of visual noise in this error message. The `FromResidual` and `Yeet` are implementation details that are not relevant to the user,
+but the relevant details are somewhat obscured. 
+We're not making a good first impression here.
+
+**And all we did was use an `Option`!**
+
+My main gripe with this error message is that it doesn't explain **why** the `?` operator doesn't work with `Option`...
+just that it doesn't.
+Granted, the error message is already quite long and it might not be the best place to teach Rust, but it would be nice to have a link to a more detailed explanation. After all, this is Rust: we take great pride in our error messages.
+
+By the way, running `rustc --explain E0277` only gives you a very generic explanation about types which don't implement traits. 
+That's not very actionable.
+
+## What Do People Usually Do?
+
+The most common solution I see is that people are confused for a bit,
+try to understand the error message, and eventually just give up, use `unwrap`,
+and make a mental note to come back to it later.
+
+```rust
+fn get_user_name() -> Result<String, String> {
+    let user = get_user().unwrap();
+    // Do something with `user`
+    Ok(user)
+}
+```
+
+Even with a trainer around, they are often too embarrassed to ask for help.
+They think people are supposed to "get this" and they are the only ones who don't.
+
+This just defers the problem. 
+The user of the function will get a panic at runtime. It might be your future self.
+
+Moreover, I advise people to avoid `unwrap` in production code, but this sets a bad example.
```

Comment:
```suggestion
Moreover, I advise people to avoid `unwrap` in production code, as this sets a bad example.
```
Not quite sure what you'd mean with "but"